### PR TITLE
Add some debugging docs

### LINKED ISSUE #232: Unit tests and debug tools
When these charts work, they are fine. When they don't work, the user is left in a world of hurt, rummaging round in a complex system trying to find clues as to what has gone wrong.

For a production system intended for others to use, these charts should come with a set of unit tests and debug tools that can be run on a cluster to check all the components are present and correct. 

A simple library of `kubectl` commands to look for obvious things would be a useful start. So rather than posting suggestions in a Slack channel, StackHPC can point people at a page of common debug queries they can copy and paste.

These scripts should include `kubectl` queries to resolve the random Pod names so that they will _'just work'_ without requiring the user to resolve the names manually.

These will be useful for people trying to port the Helm charts to new platforms, and for system admins trying to verify if their Openstack platform is working correctly.

### PR DESCRIPTION
**Pull request by sd109**
Closes #232

### CODE REVIEW

File: charts/openstack-cluster/DEBUGGING.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  W -->> M: Wait for workload cluster API server to be online
+  M ->> W: Install addons (helmreleases & manifests)
+  activate W
+  W -->> M: Wait for core addons (e.g. CNI) to be ready
+  M ->> O: Create remaining control plane and worker nodes
+  O ->> W: Create remaining servers
+  W -->> M: Wait for all addons to be ready
+  deactivate W
+  M -) U: Cluster ready
+```
```
Comment (sd109) [inline charts/openstack-cluster/DEBUGGING.md:30]: I'm still on the fence about whether this diagram is really useful here, happy to remove it depending on others' opinions.

File: README.md [hunk 2]
```diff
@@ -25,3 +25,9 @@ Currently, the following charts are available:
 | [cluster-addons](./charts/cluster-addons) | Deploys addons into a Kubernetes cluster, e.g. CNI. |
 | [etcd-defrag](./charts/etcd-defrag/) | Installs a `CronJob` for running [etcd defragmentation](https://etcd.io/docs/v3.5/op-guide/maintenance/#defragmentation). |
 | [openstack-cluster](./charts/openstack-cluster) | Deploys a Kubernetes cluster on an OpenStack cloud. |
+
+# Usage
+
+## OpenStack
+
+See the openstack-cluster chart [README.md](./charts/openstack-cluster/README.md) for deployment instructions and [DEBUGGING.md](./charts/openstack-cluster/DEBUGGING.md) for debugging tips.
```
Comment (mkjpryor) [inline README.md:33]: Don't know if this is required? I think linking the debugging page from the `openstack-cluster` README would be better?

File: charts/openstack-cluster/DEBUGGING.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
+# Debugging
```
Comment (mkjpryor) [inline charts/openstack-cluster/DEBUGGING.md:1]: Maybe "Debugging a Cluster API cluster on OpenStack"?

File: charts/openstack-cluster/DEBUGGING.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+Commonly encountered issues (usually visible in the CAPO controller logs) include:
+
+- Insufficient quota in the target OpenStack project (for servers, volumes etc.)
+
+- Network routing issues between the CAPI management cluster and the workload cluster
+
+### Essential Addons
+
+Now that the first control plane node has initialised successfully, the kubeconfig for the workload cluster will be written to a Kubernetes `Secret` named `<cluster-name>-kubeconfig` on the CAPI management cluster. At this point, the kubeconfig can be extracted from the secret for use with `kubectl` using the following command:
```
Comment (mkjpryor) [inline charts/openstack-cluster/DEBUGGING.md:136]: This is slightly misleading maybe? The `kubeconfig` secret is actually generated once the API server address is known, which is usually once the LB is created and the FIP attached (if requested).

File: charts/openstack-cluster/DEBUGGING.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+```
+kubectl --kubeconfig capi-mgmt get secret <cluster-name>-kubeconfig -o jsonpath='{.data.value}' | base64 -d > workload
```
Comment (mkjpryor) [inline charts/openstack-cluster/DEBUGGING.md:139]: The `go-template` format for `kubectl` supports a `base64decode` filter which I think is nicer than this:

```sh
kubectl --kubeconfig capi-mgmt get secret <cluster-name>-kubeconfig -o go-template='{{.data.value | base64decode}}' > workload
```

Not massively bothered either way though.

**Review by mkjpryor — CHANGES_REQUESTED** [review]
This is a good first crack.

I would maybe just be really explicit and add some examples of accessing the component logs. If you use the `deploy/<>` format, then it can be generic, e.g.:

```sh
kubectl -n capo-system logs deploy/capo-controller-manager`
```

or 

```sh
kubectl -n capi-addon-system logs deploy/cluster-api-addon-provider
```

**Review by mkjpryor — APPROVED** [review]
LGTM

**Pull request merged**